cmd/util/version.go

```diff
@@ -24,15 +25,22 @@ type Versions struct {
 }
 
 func CheckVersion(cmd *cobra.Command, args []string) error {
+	strictVersioning, err := config.Get[bool](types.NodeStrictVersionMatch)
+	if err != nil {
+		return fmt.Errorf("DEVELOPER ERROR: please file an issue with this error message: %w", err)
+	}
```

Comment:
Is there any way we can have an error-free version of config.Get?  I can see why the errors are currently returned, but it might be better for us to have an interface similar to hash where like `Get[T](string) (T, bool)` so we can flag the non-existence without raising an error that we then have to handle.  

If we pass the wrong key here it's a developer error rather than a runtime error so even better would be a compile-time error.